### PR TITLE
Feature/list

### PR DESCRIPTION
**Pull request by Andrekra**

### CODE REVIEW

File: lib/buckit/cli.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      cmd.program_name = File.basename $0
+      cmd.program_version = Buckit::VERSION
+
+      cmd.options = CmdParse::OptionParserWrapper.new do |opt|
+        opt.separator "Global options:"
+      end
+
+      cmd.main_command.short_desc = 'Tool belt for managing your S3 buckets'
+      cmd.main_command.description =<<END.strip
+Buckit provides a list of commands that will allow you to manage your content
```
Comment (johanb) [inline lib/buckit/cli.rb]: mange the content ?

File: lib/buckit/cli/create.rb [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+      def run s3, bucket, args
+        raise WrongUsage.new(nil, "You need to supply a bucket name") if not bucket
+
+        # Create bucket
+        create_bucket(s3, bucket)
+        credentials = create_user(bucket)
+        create_policy(bucket)
+        add_cors(s3, bucket)
+
+        puts "FOG_DIRECTORY=#{bucket} FOG_PROVIDER=AWS FOG_REGION=#{@region} ASSET_HOST=#{Aws::S3::Resource.new.bucket(bucket).url} AWS_ACCESS_KEY_ID=#{credentials.access_key_id} AWS_SECRET_ACCESS_KEY=#{credentials.secret_access_key}"
```
Comment (johanb) [inline lib/buckit/cli/create.rb:34]: Makes it less universal

File: lib/buckit/cli/create.rb [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+      end
+
+      protected
+      def create_bucket s3, bucket
+        s3.create_bucket(
+          acl: "public-read",
```
Comment (johanb) [inline lib/buckit/cli/create.rb]: all buckits are public by default ?

File: lib/buckit/cli/create.rb [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+          bucket: bucket
+        )
+      end
+
+      def create_user bucket
+        begin
+          iam = Aws::IAM::Client.new
+          iam.create_user(user_name: bucket, path: '/apps/')
```
Comment (johanb) [inline lib/buckit/cli/create.rb:48]: Why the "apps" path ? seems to be hidden business logic

File: lib/buckit/cli/create.rb [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+          credentials = iam.create_access_key(user_name: bucket)
+          credentials.first.access_key
+        rescue Aws::IAM::Errors::ServiceError
+          raise FailureFeedback.new("Failed creating IAM")
+        end
+      end
+
+      def create_policy bucket
```
Comment (johanb) [inline lib/buckit/cli/create.rb]: Misleading name, this method doesn't just create a policy, it creates a policy and applies it to a user, which is implicitly tied to the bucket name

File: lib/buckit/version.rb [hunk 6]
```diff
@@ -1,3 +1,3 @@
 module Buckit
-  VERSION = "0.0.3"
+  VERSION = "1.0.0"
```
Comment (johanb) [inline lib/buckit/version.rb:2]: Debatable

File: lib/buckit/cli/create.rb (hunk 3 above)
Comment (Andrekra) [inline lib/buckit/cli/create.rb]: Yeah, same as previous buckit command and our common use case. 99% of every bucket we create are public readable.  https://github.com/wirelab/buckit/blob/develop/bin/buckit#L15

File: lib/buckit/cli/create.rb (hunk 4 above)
Comment (Andrekra) [inline lib/buckit/cli/create.rb:48]: No specific reason. Easy categorization of users in the console. Atm we have wirelab users and apps in 1 massive list. I can remove this

File: lib/buckit/cli/create.rb (hunk 3 above)
Comment (johanb) [inline lib/buckit/cli/create.rb]: Could be, but lets make it a configurable option with default to public

File: lib/buckit/cli/base_cmd.rb [hunk 7]
```diff
@@ -0,0 +1,59 @@
+require 'buckit/exceptions'
+require 'aws-sdk'
+require 'cmdparse'
+
+module Buckit
+  module CLI
+    class BaseCmd < CmdParse::Command
+      @has_prefix = false
```
Comment (Darksecond) [inline lib/buckit/cli/base_cmd.rb:8]: wist niet dat dit zo werkte

**Pull request merged**